File: ALGORITHMS/C++/bubbleSort.cpp
#include <iostream>
using namespace std;

void BubbleSort(int* tab, int size){

    for(int i =0; i<size-1;i++){
        for(int j =0; j<size-i-1;j++){
            if(tab[j]>tab[j+1]){
                int temp = tab[j];
                tab[j]=tab[j+1];
                tab[j+1]=temp;
            }

        }
    }

}



int main(){

    int tab[]={3,12,2,45,1333,54,23};
    int size = sizeof(tab)/sizeof(tab[0]);

    
    cout<<"Before: "<<endl;
    for(int i : tab){
        cout<<i<<" ";
    }
    cout<<endl;

    cout<<"After: "<<endl;
    BubbleSort(tab,size);
    for(int i : tab){
        cout<<i<<" ";
    }




}

File: ALGORITHMS/PYTHON/bubble_sort.py
#bubble_sort

def bubbleSort(tab):  
    for i in range(n):
        for j in range(0, n - i - 1):
            if tab[j] > tab[j + 1]:  
                temp = tab[j]
                tab[j] = tab[j + 1]
                tab[j + 1] = temp


def addNumbers(tab,n):
    for i in range(n):
        print(f"give me the numbers")
        number = int(input(f"number{i}: "))
        tab.append(number)


n = 5
tab = [n]


addNumbers(tab,n)
print(f"before: \n{tab}")
bubbleSort(tab)  
print(f"after: \n{tab}")

